app/src/main/java/com/randomappsinc/studentpicker/utils/JSONUtils.kt

```diff
@@ -0,0 +1,74 @@
+package com.randomappsinc.studentpicker.utils
+
+import android.text.TextUtils
+import com.randomappsinc.studentpicker.models.ListDO
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+
+object JSONUtils {
+
+    private const val NAME_LIST = "nameList"
+    private const val LIST_NAME = "listName"
+    private const val NAME = "name"
+    private const val AMOUNT = "amount"
+
+    @JvmStatic
+    fun namesArrayToJsonString(chosenNames: List<String>): String {
+        val alreadyChosenNamesArray = JSONArray()
+        for (alreadyChosenName in chosenNames) {
+            alreadyChosenNamesArray.put(alreadyChosenName)
+        }
+        return alreadyChosenNamesArray.toString()
+    }
+
+    @JvmStatic
+    fun extractNamesHistory(namesArrayText: String?): List<String> {
+        if (TextUtils.isEmpty(namesArrayText)) {
+            return emptyList()
+        }
+        val alreadyChosenNames: MutableList<String> = ArrayList()
+        try {
+            val namesArray = JSONArray(namesArrayText)
+            for (i in 0 until namesArray.length()) {
+                alreadyChosenNames.add(namesArray.getString(i))
+            }
+        } catch (ignored: JSONException) {
+        }
+        return alreadyChosenNames
+    }
+
+    fun serializeNameList(listDOs: List<ListDO>): String {
+        val nameListArray = JSONArray()
+        for (listDO in listDOs) {
+            val listDOJson: JSONObject? = createNameListJson(listDO)
+            if (listDOJson != null) {
+                nameListArray.put(listDOJson)
+            }
+        }
+        return nameListArray.toString()
+    }
+
+    private fun createNameListJson(listDO: ListDO): JSONObject? {
+        try {
```

Comment:
The is no need to try and catch as JSON as the compiler compile the code without any issues and from what I remember Kotlin doesn't have checked exceptions